collections/_article/uid-changes-coming-to-godot-4-4.md

```diff
@@ -0,0 +1,87 @@
+
+### Why not use a single centralized file to store all the project's UIDs?
+
+Many other programs use a centralized database to store UIDs for all files in the project. While this approach makes for a leaner filesystem, it also has several downsides:
+
+- This file would be subject to frequent merge conflicts in version control systems. For example, if you have two branches that add or edit different scripts/shaders, they would each add their own lines to the UID database file. Both of these branches would add these lines at the same location in the file (probably at the end). However, once you try to merge those two branches together, your version control system wouldn't be able to know what to keep automatically. This is where a merge conflict occurs.
+  - In contrast, using separate files ensures merge conflicts don't happen quite as often. If conflicts do arise, they can be resolved by just including the file (or removing it if the source file has also been removed).
+- Centralizing UID information is problematic for add-ons. For example, if a developer refactors their addon and moves files around, then someone downloads the new version of their addon into an existing project of theirs, references to files in the addon in their project get updated automatically. If UIDs are stored in a centralized fashion, there wouldn't be a way to do this as the addon's UIDs wouldn't be stored in the addon. This would result in UID references breaking.
+- If you have multiple Godot editor instances open on the same project, each instance may write a different version of the file, leading to conflicts that can only be detected once you reopen the editor. Additionally, on Windows, you would get a lot of errors while using the editor due to file locking limitations.
+
+While using separate UID files is not perfect by any means, we felt it was a better compromise for Godot's design and to ensure version control remains as seamless as possible.
+
+### What happens if I don't commit `.uid` files to version control?
+
+If you forget to commit `.uid` files to version control, the project will still work locally. However, as soon as you clone the project on another device, the UID references will break. Godot will still be able to resolve them if the paths haven't changed (since paths are also saved as a fallback), but a warning will be printed for every resource where a path fallback had to be used to infer the reference. This behavior should not be relied upon intentionally, as it is possible for two resources to be swapped around without any of their paths changing. This can occur if one of the resources' paths was changed to be identical to another resource that was relocated elsewhere.
```

Comment:
```suggestion
If you forget to commit `.uid` files to version control, the project will still work locally. However, as soon as you clone the project on another device, the UID references will break. Godot will still be able to resolve them if the paths haven't changed (since paths are also saved as a fallback), but a warning will be printed for every resource where a path fallback had to be used to infer the reference. This behavior should not be relied upon intentionally, as it is possible for two resources to be swapped around without either of their paths changing. This can occur if the path of one of the resources was changed to be identical to another resource that was relocated elsewhere.
```